gcc/rust/typecheck/rust-tyty-resolver.h

```diff
@@ -63,9 +64,25 @@ class TyTyResolver
 	  TyTy::TyBase *resolved = nullptr;
 	  if (!context->lookup_type (hir_node_ref, &resolved))
 	    {
-	      rust_fatal_error (mappings->lookup_location (hir_node_ref),
-				"failed to lookup type for reference");
-	      return false;
+	      // this could be an array/adt type
```

Comment:
this pice of code will get abstracted more with tuples it a bit hacky at the moment. Its the code responsible to handling the case of:

let mut x;
x = 1;

so inference variables get resolved correctly.